x-pack/plugins/security/server/routes/session_management/delete_all.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import typeDetect from 'type-detect';
+import { schema } from '@kbn/config-schema';
+import { RouteDefinitionParams } from '..';
+
+/**
+ * Defines routes required for the deleting of all sessions.
+ */
+export function defineDeleteAllSessionsRoutes({ router, getSession }: RouteDefinitionParams) {
+  router.delete(
+    {
+      path: '/internal/security/session/_all',
+      validate: {
+        query: schema.maybe(
```

Comment:
__note:__ I was also considering `POST` + `body` + `/internal/security/session/_invalidate`, but didn't find enough benefits comparing to `DELETE` + `query`. I know [some ES APIs](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-invalidate-token.html) use `body` with `DELETE`, but IIRC it's not widely recommended and some proxies/tools may not be happy about it (e.g. the REST tool I use :slightly_smiling_face: ).

 Let me know what approach you prefer.

Reply:
I see you've noted in the PR description that this is an internal route for now. Is there a reason not to document this as a public endpoint? We intend for this to be used externally, and marking it as public won't preclude us from making a breaking change in a minor release if we need to. In other words, it could be an experimental/beta public API, much like our SO, Space, and Role APIs.

Reply:
> The _all suffix we're using above confused me initially -- at first glance, I assumed it would always invalidate all sessions, but we also accept a query parameter to optionally invalidate a subset of sessions.

Yeah, that's because I added filter at a later iteration :slightly_smiling_face: Agree it's confusing now.

> and marking it as public won't preclude us from making a breaking change in a minor release if we need to. In other words, it could be an experimental/beta public API, much like our SO, Space, and Role APIs.

That's a good point, I completely forgot about the experimental APIs! The "no guarantees yet" was my main motivator behind keeping this API as internal.